pkg/registry/core/pod/storage/eviction.go

```diff
@@ -307,12 +308,20 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 	return &metav1.Status{Status: metav1.StatusSuccess}, nil
 }
 
-func addConditionAndDeletePod(r *EvictionREST, ctx context.Context, name string, validation rest.ValidateObjectFunc, options *metav1.DeleteOptions) error {
-	if feature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
-		pod, err := getPod(r, ctx, name)
-		if err != nil {
-			return err
+func addConditionAndDeletePod(r *EvictionREST, ctx context.Context, name string, pod *api.Pod, deleteOptions *metav1.DeleteOptions) error {
+	var err error
+	deleteOptionsCopy := deleteOptions.DeepCopy()
```

Comment:
I'd suggest copying this in place and doing it inside the `if` block only if we need to update conditions


```go
if !dryrun.IsDryRun(deleteOptionsCopy.DryRun) && feature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
  options = options.DeepCopy()
  ...
```

Reply:
+1, moved only to the place where we need to change the options